src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+import {EventEmitter} from "events";
+
+// copy the pg.Result type: https://node-postgres.com/api/result
+export interface IPostgresResult {
+    rowCount: number;
+    command: string;
+}
+
+export interface IPostgresData {
+    query: {
+        text?: string;
+        plan?: string;
+        preparable?: {
+            text: string;
+            args: any[];
+        }
+    };
+    database: {
+        host: string;
+        port: string;
+    };
+    result?: IPostgresResult;
+    duration: number;
+    error?: Error;
+}
+
+type PostgresCallback = (err: Error, res: IPostgresResult) => any;
+
+function postgres6PatchFunction(originalPg, originalPgPath) {
+    const originalClientQuery = originalPg.Client.prototype.query;
+
+    // this keeps track of user-supplied callback => patched callbacks so we don't keep
+    // re-patching the same callback over and over for library functions
+    // IMPORTANT: use a WeakMap so that we don't keep references to temporary/otherwise unreferenced functions
+    const patchedCallbacks = new WeakMap<PostgresCallback, PostgresCallback>();
```

Comment:
Ah, I suppose supporting ES5 is important for node < 6, I didn't even think of that. I was using that originally but I thought the weakmap would be more clever.